### ISSUE TITLE
Bookmark tagging

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#1057]
Carried over from https://github.com/brave/browser-laptop/issues/5078

## Description 
> Large collections of bookmarks would be easier to manage if bookmarks could be tagged and were subsequently searchable by those tags

**Comment by eljuno** [comment]
+1 from Community https://community.brave.com/t/bookmarks-tags-in-brave/46645?u=eljuno

**Comment by Taariqq** [comment]
This was one thing that made me switch back to Firefox recently. Brave bookmarks are ancient.
I have solved other requirements switching recent tabs and vertical sidebar, with extensions .. but ... being able to search bookmarks based on tags is ... well, the last thing for me.

Last night I had both Brave and Firefox open with multiple tabs. For some reason Firefox decided to become a hog and rendered my usage of the computer almost useless. Had to come back to Brave.

Hoping to either go back to Firefox asap or seeing the hash-tags in bookmarks in Brave. I rely on them instead of using Pocket and stuff.

**Comment by jonathansampson** [comment]
+1 from [a user on Twitter](https://twitter.com/vastidity/status/1245811820931117056) recently.

**Comment by yet-another-tim** [comment]
+1 tag assignment to bookmarks and searching bookmarks by tag. Currently use Firefox on Linux. Tried switching to Brave. Happy with performance, privacy, and web extension access (Firefox falls behind here). Ultimately switched back to Firefox due to bookmark tagging. Was deal breaker to go without.

**Comment by antonio-sn** [comment]
I need this feature in order to move to Brave.

**Comment by Emirov** [comment]
This feature is so important to manage bookmarks. In fact, there are some bookmarks that belongs to more than one subject, so putting them in one folder doesn't suffice.  Also this option exists in other browser. I think this issue can even deprive Brave to be the number for me despite thr other excellent features that it offers.

**Comment by kaushalyap** [comment]
@BrendanEich Can we please get this done.

Please implement bookmark tagging this has been [requested in Chromium about a decade ago](https://bugs.chromium.org/p/chromium/issues/detail?id=17536), it seems that they do not care. Bookmark tagging had been their in Firefox for a long time. For Chromium based browser we have to depend to third party extensions to do the tagging, which complicate things.

**Comment by samjoy** [comment]
@BrendanEich Can you please add bookmark tagging? It would be really helpful in managing bookmarks.

**Comment by Coldfusionstorm** [comment]
Yes. This feature is sorely needed, Firefox has really gone downhill in recent years and i only keep coming back to Firefox because i can't manage my bookmarks in brave properly.(Notice how the fox is pretty much gone in the firefox icon)

Just keeping a bookmark in folder kind of reminds me of opening the bookmark sidebar in internet explorer.

That's just not a nice feeling.

**Comment by danielb2** [comment]
I'll donate 💰 for this implementation. I love that firefox has this. It's been a decade and it's still not in chrome. It makes organizing so much better

**Comment by john-soda** [comment]
For me bookmark tags are also the last piece in the puzzle before I can completely switch to brave. But without this feauture is not possible because this bookmark tags are too important for me.

**Comment by thalskarth** [comment]
It's also important to have the option to import the tags from firefox. 

> @BrendanEich Can we please get this done.
> 
> This has been [requested in Chromium about a decade ago](https://bugs.chromium.org/p/chromium/issues/detail?id=17536), it seems that they do not care. Bookmark tagging had been their in Firefox for a long time. For Chromium based browser we have to depend to third party extensions to do the tagging, which complicate things.

Do those extensions import from firefox's tags? Could you give a link to them?

**Comment by rickalex21** [comment]
This is the only thing stopping me from moving to brave. I love tags on firefox.

**Comment by dpthegrey** [comment]
Having bookmark tags will be very helpful in searching and sorting bookmarks and be able to filter bookmarks based on tags.

**Comment by InterruptDriven** [comment]
Brave...Vivaldi...Brave...Vivaldi: which ever browser is able to completely replicate the bookmark tag/metadata facility of Firefox will become my primary browser.

**Comment by inson1** [comment]
+1

**Comment by forvenus** [comment]
bookmark + tag + easy UI can replace raindrop/pocket....!

**Comment by unknownterritory** [comment]
I, too, would love to switch to Brave as my default browser and this remains the only issue that holds me back. Taxonomy is the only way of properly cataloguing any large collection of items, it should be obvious. It's puzzling to me that this is not a priority for most browsers.

**Comment by unknownterritory** [comment]
If you haven't yet, maybe vote on this feature: https://community.brave.com/t/bookmarks-tags-in-brave/46645

**Comment by ted-moke** [comment]
I wrote a script to extract the tags from your exported FF bookmarks and append to the title of the bookmark.

[https://github.com/ted-moke/export-ff-bookmark-tags](https://github.com/ted-moke/export-ff-bookmark-tags)

**Comment by c-j-mills** [comment]
I completely agree with this feature.

Coming from Firefox to Brave, this feature is critical to organizing many bookmarks—the ability to tag a bookmark with one or more tags permits ease of organization and search.

In addition, placing a Tag (rather than a Bookmark) into the Bookmark Toolbar enables dynamic folders of tagged bookmarks. For example, simply tagging a bookmark that appears in the Bookmarks Toolbar's Tag with no further configuration is a superpower. Firefox has had this ability for some time and using the Brave's current bookmarks system is a little painful.

**Comment by rickalex21** [comment]
Not sure if this feature is ever going to happen. This has been open for 7 years now.

**Comment by t0ma5** [comment]
Total deal breaker for me 🥲

**Comment by yusuphgammut** [comment]
I'd really love to see this feature in Brave!

**Comment by JChenJChen** [comment]
bump

**Comment by okonon** [comment]
This feature would be awesome to have!

**Comment by venuvoldeos** [comment]
This is th only reason holding me back from migrating to Brave from Firefox completely. It makes day to day tasks very easy.
